packages/create-llama/helpers/index.ts

```diff
@@ -95,22 +95,34 @@ const createEnvLocalFile = async (
   }
 };
 
-const generateContextData = async (
+// eslint-disable-next-line max-params
+async function generateContextData(
   framework: TemplateFramework,
   packageManager?: PackageManager,
   openAiKey?: string,
   vectorDb?: TemplateVectorDB,
-) => {
+  dataSource?: TemplateDataSource,
+  llamaCloudKey?: string,
+) {
   if (packageManager) {
     const runGenerate = `${cyan(
       framework === "fastapi"
         ? "poetry run python app/engine/generate.py"
         : `${packageManager} run generate`,
     )}`;
     const hasOpenAiKey = openAiKey || process.env["OPENAI_API_KEY"];
+    const hasLlamaCloudKey = (dataSource?.config as FileSourceConfig)
+      ?.useLlamaParse
+      ? llamaCloudKey || process.env["LLAMA_CLOUD_API_KEY"]
+      : true;
     const hasVectorDb = vectorDb && vectorDb !== "none";
     if (framework === "fastapi") {
-      if (hasOpenAiKey && !hasVectorDb && isHavingPoetryLockFile()) {
+      if (
+        hasOpenAiKey &&
+        hasLlamaCloudKey &&
```

Comment:
how about naming it like this?
```suggestion
        openAiKeyConfigured &&
        llamaCloudKeyConfigured &&
```

Reply:
It's nice, thanks for the suggestion. I'm updating it, naming is hard :D